nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -462,178 +164,295 @@ def forward(self, obj):
-            return torch.cat(args, dim=self.cat_dim)
+    return my_partial(func, positional, keyword, undetermined, undetermined_special_treat)
 
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    dim = inputs[1].toIValue()
-    return CatModule(dim)
-
-
-def expandas_python(node, speedup):
-    class ExpandasModule(torch.nn.Module):
-        def forward(self, x, y):
-            return x.expand_as(y).clone()
-    return ExpandasModule()
-
-
-trans_from_jit_to_python = {
-    'aten::add': add_python,
-    'aten::add_': add_python,
-    'aten::sub': sub_python,
-    'aten::sub_': sub_python,
-    'aten::mul': mul_python,
-    'aten::mul_': mul_python,
-    'aten::relu': relu_python,
-    'aten::relu_': relu_inplace_python,
-    'aten::sigmoid': sigmoid_python,
-    'aten::sigmoid_': sigmoid_python,
-    # tanh behaives like relu
-    'aten::tanh': relu_python,
-    'aten::tanh_': relu_python,
-    'aten::flatten': flatten_python,
-    'aten::mean': mean_python,
-    'aten::dropout': dropout_python,
+trans_func_dict = {
     'aten::slice': slice_python,
-    'aten::select': select_python,
-    'aten::size': size_python,
-    'aten::t': transpose_python,
-    'aten::transpose': transpose2_python,
-    'aten::Int': toint_python,
-    'aten::view': view_python,
-    'aten::reshape': reshape_python,
-    'aten::permute': permute_python,
-    'aten::matmul': matmul_python,
-    'aten::div': div_python,
-    'aten::floor_divide': floor_div_python,
-    'aten::softmax': softmax_python,
-    'aten::contiguous': contiguous_python,
-    'aten::gelu': gelu_python,
-    'aten::cat': cat_python,
-    'aten::avg_pool2d': avgpool2d_python,
-    'aten::max_pool2d': avgpool2d_python,
-    'aten::adaptive_avg_pool2d': adaptive_avgpool_python,
-    'aten::to': to_python,
-    'aten::type_as': typeas_python,
-    'aten::upsample_bilinear2d': upsample_bilinear2d_python,
-    'aten::upsample_nearest2d': upsample_nearest2d_python,
-    'aten::exp': exp_python,
-    'aten::squeeze': squeeze_python,
-    'aten::unsqueeze': unsqueeze_python,
-    'aten::constant_pad_nd': constant_pad_nd_python,
-    'aten::silu': silu_python,
-    'aten::expand_as': expandas_python,
+    'aten::Int': partial(generate_aten_to_python, torch._C._TensorBase.int),
     'prim::TupleUnpack': tupleunpack_python,
     'prim::ListUnpack': tupleunpack_python,
     'prim::NumToTensor': num2tensor_python,
-    'prim::GetAttr': getattr_python
+    'prim::GetAttr': getattr_python,
 }
+def init_add_functions(func_from: ModuleType | Type):
+    """
+    Add function/method attributes from a module/class, to the trans_func_dict
 
+    Parameters
+    ---------
+    func_from: module/class
+        The module/class include needed functions
 
-def jit_to_python_function(node, speedup):
+    """
+    global trans_func_dict
+    new_trans_func_dict = dict()
+    for name in dir(func_from):
+        attr = getattr(func_from, name)
+        if callable(attr) and not name.startswith("__"):
```

Comment:
In NNI we generally prefer single quote.

Reply:
sorry, which line?

Reply:
`startswith("__")` -> `startswith('__')`